### PR TITLE
feat(CC): add a datasource to get the list of CC bandwidth packages

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add a datasource to get the list of CC bandwidth packages
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add a datasource to get the list of CC bandwidth packages
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/cc" TESTARGS="-run TestAccDataSourceCcBandwidthPackages_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cc -v -run TestAccDataSourceCcBandwidthPackages_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceCcBandwidthPackages_basic
=== PAUSE TestAccDataSourceCcBandwidthPackages_basic
=== CONT  TestAccDataSourceCcBandwidthPackages_basic
--- PASS: TestAccDataSourceCcBandwidthPackages_basic (53.59s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cc        53.641s

```

### CODE REVIEW

File: docs/data-sources/cc_bandwidth_packages.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `bandwidth_package_id` - (Optional, String) Specifies the bandwidth package ID.
+
+* `name` - (Optional, String) Specifies the bandwidth package name.
+
+* `enterprise_project_id` - (Optional, String) Specifies ID of the enterprise project that the bandwidth package
```
Comment (houpeng80) [inline docs/data-sources/cc_bandwidth_packages.md]:
```suggestion
* `enterprise_project_id` - (Optional, String) Specifies the ID of the enterprise project that the bandwidth package
```

File: docs/data-sources/cc_bandwidth_packages.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+* `billing_mode` - (Optional, String) Specifies the billing mode of the bandwidth package.
+  The options are as follows:
+  + **1**：pay by period for the Chinese Mainland website.
+  + **2**: pay by period for the International website.
+  + **3**: pay-per-use for the Chinese Mainland website.
+  + **4**: pay-per-use for the International website.
+  + **5**: 95th percentile bandwidth billing for the Chinese Mainland website.
+  + **6**: 95th percentile bandwidth billing for the International website.
+
+* `resource_id` - (Optional, String) Specifies ID of the resource that the bandwidth package is bound to.
```
Comment (houpeng80) [inline docs/data-sources/cc_bandwidth_packages.md]:
```suggestion
* `resource_id` - (Optional, String) Specifies the ID of the resource that the bandwidth package is bound to.
```

File: huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_bandwidth_packages_test.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  name           = "%[1]s2"
+  local_area_id  = "Chinese-Mainland"
+  remote_area_id = "Chinese-Mainland"
+  charge_mode    = "bandwidth"
+  billing_mode   = 3
+  bandwidth      = 12
+  description    = "desc 2"
+		
+  tags = {
+    foo   = "bar"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_bandwidth_packages_test.go]:
```suggestion
    foo = "bar"
```

File: huaweicloud/services/cc/data_source_huaweicloud_cc_bandwidth_packages.go [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+
+	rst := make([]interface{}, 0, len(curArray))
+
+	if bandwidthFilter, ok := d.GetOk("bandwidth"); ok {
+		for _, item := range curArray {
+			itemMap := item.(map[string]interface{})
+			if int(itemMap["bandwidth"].(float64)) != bandwidthFilter {
+				continue
+			}
+			rst = append(rst, item)
```
Comment (houpeng80) [inline huaweicloud/services/cc/data_source_huaweicloud_cc_bandwidth_packages.go]:
```suggestion
			if int(itemMap["bandwidth"].(float64)) == bandwidthFilter {
				rst = append(rst, item)
			}
```

File: docs/data-sources/cc_bandwidth_packages.md (hunk 1 above)
Comment (profoundwu) [reply docs/data-sources/cc_bandwidth_packages.md]: fixed

File: huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_bandwidth_packages_test.go (hunk 3 above)
Comment (profoundwu) [reply huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_bandwidth_packages_test.go]: fixed

File: huaweicloud/services/cc/data_source_huaweicloud_cc_bandwidth_packages.go (hunk 4 above)
Comment (profoundwu) [reply huaweicloud/services/cc/data_source_huaweicloud_cc_bandwidth_packages.go]: fixed.

File: docs/data-sources/cc_bandwidth_packages.md (hunk 2 above)
Comment (profoundwu) [reply docs/data-sources/cc_bandwidth_packages.md]: fixed.

**Pull request merged**